devU-api/src/nonContainerAutoGrader/nonContainerAutoGrader.router.ts

```diff
@@ -0,0 +1,53 @@
+// Libraries
+import express from 'express'
+
+// Middleware
+import validator from './nonContainerAutoGrader.validator'
+import {asInt} from '../middleware/validator/generic.validator'
+
+// Controller
+import nonContainerQuestions from './nonContainerAutoGrader.controller'
+
+const Router = express.Router()
+
+/**
+ * @swagger
+ * /nonContainerAutoGrader:
+ *   get:
+ *     summary: Retrieve a list of all nonContainerAutoGrader with the assignment ID
+ */
+Router.get('/:assignmentId', nonContainerQuestions.get)
+
+/**
+ * @swagger
+ * /nonContainerAutoGrader/{id}:
+ *   get:
+ *     summary: Retrieve a single question
+ */
+Router.get('/:id', asInt(), nonContainerQuestions.detail)
```

Comment:
There's nothing to distinguish these two GET paths. I'm not sure what is the best way to design these paths, but something like "/nonContainerAutoGrader/bySubmission/:submissionId" would work.. maybe just "/nonContainerAutoGrader/submission/:submissionId"

We should also still have GET "/nonContainerAutoGrader" that lists every single non-container auto grader. I know we discussed this and we def want a path that lists by assignment, but I can imagine some admin/data analysis use cases where we do want everything. It also gives us a way to retrieve graders for assignments that have been deleted without having to go to the DB directly

Reply:
I don't belive NonContainerAutoGrader has access to the submission id,

these are the paths I came up with 

for getting all question by Assignment ID

```
 /nonContainerAutoGrader/byAssignment/{assignmentId}:
    get:
     summary: Retrieve a list of all nonContainerAutoGraders with the assignment ID
 ```
 
for getting a specific question

 ```
 /nonContainerAutoGrader/byId/{id}:
    get:
     summary: Retrieves a single question by ID
 ```
 
 for getting all questions

```
 /nonContainerAutoGrader:
    get:
      summary: get all NonContainerAutoGraders questions 
 ```